lib/wibox/container/arcchart.lua

```diff
@@ -0,0 +1,317 @@
+        ) }
+    end
+end
+
+-- Fit this layout into the given area
+function arcchart:fit(context, width, height)
+    if self._private.widget then
+        local wa = content_workarea(self, width, height)
+        return base.fit_widget(self, context, self._private.widget, wa.width, wa.height)
+    end
+
+    return width, height
+end
+
+--- The widget to wrap in a radial proggressbar.
+-- @property widget
+-- @tparam widget widget The widget
+
+function arcchart:set_widget(widget)
+    if widget then
+        base.check_widget(widget)
+    end
+    self._private.widget = widget
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Get the number of children element
+-- @treturn table The children
+function arcchart:get_children()
+    return {self._private.widget}
+end
+
+--- Replace the layout children
+-- This layout only accept one children, all others will be ignored
+-- @tparam table children A table composed of valid widgets
+function arcchart:set_children(children)
+    self._private.widget = children and children[1]
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Reset this layout. The widget will be removed and the rotation reset.
+function arcchart:reset()
+    self:set_widget(nil)
+end
+
+for k,v in ipairs {"left", "right", "top", "bottom"} do
+    arcchart["set_"..v.."_padding"] = function(self, val)
+        self._private.paddings = self._private.paddings or {}
+        self._private.paddings[v] = val
+        self:emit_signal("widget::redraw_needed")
+    end
+end
+
+--- The padding between the outline and the progressbar.
+--@DOC_wibox_container_arcchart_padding_EXAMPLE@
+-- @property paddings
+-- @tparam[opt=0] table|number paddings A number or a table
+-- @tparam[opt=0] number paddings.top
+-- @tparam[opt=0] number paddings.bottom
+-- @tparam[opt=0] number paddings.left
+-- @tparam[opt=0] number paddings.right
+
+--- The progressbar value.
+--@DOC_wibox_container_arcchart_value_EXAMPLE@
+-- @property value
+-- @tparam number value Between min_value and max_value
+
+function arcchart:set_value(val)
+    if not val then self._percent = 0; return end
+
+--     if val > 0 then
+--         self:set_max_value(val)
+--     elseif val < 0 then
+--         self:set_min_value(val)
+--     end
+
+    local delta = 0 - 0
+
+    self._percent = val/delta
+    self:emit_signal("widget::redraw_needed")
+end
+
+--- The border background color.
+--@DOC_wibox_container_arcchart_border_color_EXAMPLE@
+-- @property border_color
+
+--- The border foreground color.
+--@DOC_wibox_container_arcchart_color_EXAMPLE@
+-- @property color
+
+--- The border width.
+--@DOC_wibox_container_arcchart_border_width_EXAMPLE@
+-- @property border_width
+-- @tparam[opt=3] number border_width
+
+--- The minimum value.
+-- @property min_value
+
+--- The maximum value.
+-- @property max_value
+
+for _, prop in ipairs {"border_width", "border_color", "paddings", "colors",
+    "rounded_edge", "bg", "thickness", "values", "min_value", "max_value" } do
```

Comment:
Consistency: Here it is called `min/max_value`, some previous widget called it `minimum/maximum_value` (see the comment that I left there). It would be nice if both used the same name.